crates/ironrdp-pdu/src/rdp/capability_sets/bitmap_codecs.rs

```diff
@@ -641,17 +641,77 @@ impl CodecId {
     }
 }
 
-pub fn client_codecs_capabilities() -> BitmapCodecs {
-    let codecs = vec![Codec {
-        id: CODEC_ID_REMOTEFX.0,
-        property: CodecProperty::RemoteFx(RemoteFxContainer::ClientContainer(RfxClientCapsContainer {
-            capture_flags: CaptureFlags::empty(),
-            caps_data: RfxCaps(RfxCapset(vec![RfxICap {
-                flags: RfxICapFlags::empty(),
-                entropy_bits: EntropyBits::Rlgr3,
-            }])),
-        })),
-    }];
-
-    BitmapCodecs(codecs)
+/// This function generates a list of client codec capabilities based on the
+/// provided configuration.
+///
+/// # Arguments
+///
+/// * `config` - A slice of string slices that specifies which codecs to include
+///   in the capabilities. Codecs can be explicitly turned on ("codec:on") or
+///   off ("codec:off").
+///
+/// # List of codecs
+///
+/// * `remotefx` (on by default)
+///
+/// # Returns
+///
+/// A vector of `Codec` structs representing the codec capabilities, or an error
+/// suitable for CLI.
+pub fn client_codecs_capabilities(config: &[&str]) -> Result<BitmapCodecs, String> {
+    use std::collections::HashMap;
```

Comment:
note: I think this kind of CLI-related logic does not belong to ironrdp-pdu. But I have local changes to improve config handling and support the .RDP file format. Maybe we could revisit afterwards.